src/sampling.rkt

```diff
@@ -119,6 +123,63 @@
     (timeline-push! 'method "random")
     (cons (λ () (map random-generate reprs)) (hash 'unknown 1.0))]))
 
+(define bool-discretization
+  (discretization identity
+                  (lambda (x y) (if (eq? x y) 0 1))))
+
+(define (representation->discretization repr)
+  (discretization
+   (representation-bf->repr repr)
+   (lambda (x y) (- (ulp-difference x y repr) 1))))
```

Comment:
We can modify `ulp-distance` in `float.rkt`, because now that function adds 1 - which is 1 more than the actual ulp distance, and here we subtract this 1

Reply:
Yeah, but I think then we'd need to add the `+ 1` in a lot more places. If the goal was to make this faster I think we would instead inline `ulp-difference`, then we could also pre-compute a few `repr`-related things.